components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/connector/CoordinatedTaskDBConnector.java

```diff
@@ -0,0 +1,362 @@
+            }
+        }
+    }
+
+    /**
+     * Removes the node id of the task and update the task state.
+     *
+     * @param coordinatedTasks - List of coordinated tasks which needs to be updated.
+     */
+    public void cleanInvalidNodesAndUpdateState(List<CoordinatedTask> coordinatedTasks) throws SQLException {
+
+        if (coordinatedTasks.isEmpty()) {
+            return;
+        }
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.CLEAN_INVALID_NODES)) {
+            for (CoordinatedTask task : coordinatedTasks) {
+                String taskName = task.getTaskName();
+                preparedStatement.setString(1, task.getTaskState().name());
+                preparedStatement.setString(2, taskName);
+                preparedStatement.addBatch();
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Removing the node assignment of task [" + taskName + "].");
+                }
+            }
+            preparedStatement.executeBatch();
+        }
+    }
+
+    /**
+     * For all the tasks which has this destined node id , sets it to null and update the task state to none if it was
+     * in a running state.
+     *
+     * @param nodeId - Node Id.
+     */
+    public void cleanTasksOfNode(String nodeId) throws SQLException {
+
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.CLEAN_TASKS_OF_NODE)) {
+            preparedStatement.setString(1, nodeId);
+            LOG.debug("Cleaning the tasks of node [" + nodeId + "].");
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Cleaning the tasks of node [" + nodeId + "].");
+            }
+            preparedStatement.executeUpdate();
+        }
+    }
+
+    /**
+     * Retrieves the list of tasks of node specified in specified state.
+     *
+     * @param nodeID - Id of the node, for which the tasks need to be retrieved.
+     * @param state  - State of the tasks which need to be retrieved.
+     * @return - List of tasks.
+     */
+    public List<String> retrieveTasksOfParticularNodeInSpecifiedState(String nodeID, CoordinatedTask.States state)
+            throws SQLException {
+
+        List<String> taskList = new ArrayList<>();
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.RETRIEVE_TASKS_OF_PARTICULAR_NODE)) {
+            preparedStatement.setString(1, nodeID);
+            preparedStatement.setString(2, state.name());
+            populateCoordinatedTaskNames(preparedStatement, taskList);
+        }
+        debugLogs(taskList);
+        return taskList;
+    }
+
+    private void debugLogs(List<String> tasks) {
```

Comment:
may be printDebugLogs?

Reply:
If you introduced this method just because you cannot have the same name, the solution would be to override the toString() Method in CoordinatedTask class and to have generic object list as the input parameter.